### PR TITLE
Always use env_or_config or env_or_settings to correctly handle empt…

### LINKED ISSUE #852: c2c.redis_url is never used
Documentation says that the  the redis urls can be set with `C2C_REDIS_URL` env variable or `c2c.redis_url` configuration variable but the configuration variable is never read.

### PR DESCRIPTION
**Pull request by sbrunner**
…y environment variable

fix #852 and #837

I also rename _utils to utils_ because he isn't private ...

### CODE REVIEW

File: c2cwsgiutils/profiler.py [hunk 1]
```diff
@@ -3,9 +3,11 @@
 from tempfile import gettempdir
 from typing import Any, Callable
 
+import c2cwsgiutils.utils_
+
 LOG = logging.getLogger(__name__)
-PATH = os.environ.get("C2C_PROFILER_PATH", "")
-_MODULES = os.environ.get("C2C_PROFILER_MODULES", "")
+PATH = c2cwsgiutils.utils_.env_or_settings(None, "C2C_PROFILER_PATH", None, "")
```
Comment (gberaudo) [inline c2cwsgiutils/profiler.py:9]: It looks wrong to me to use env_or_settings without passing some settings.

File: .prospector.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ pylint:
     - ungrouped-imports
     - unused-argument
     - use-symbolic-message-instead
+    - wrong-import-order
```
Comment (gberaudo) [inline .prospector.yaml:24]: Is there a good and wrong import order in python or is it only pedantic change?

File: c2cwsgiutils/redis_utils.py [hunk 3]
```diff
@@ -17,24 +18,33 @@
...
+REDIS_SERVICENAME_KEY_PROP = "c2c.redis_servicename"
+REDIS_DB_KEY_PROP = "c2c.redis_db"
+
 _master: Optional[redis.Redis] = None
 _slave: Optional[redis.Redis] = None
 _sentinel: Optional[redis.sentinel.Sentinel] = None
 
 
-def get() -> Tuple[Optional[redis.Redis], Optional[redis.Redis], Optional[redis.sentinel.Sentinel]]:
+def get(settings) -> Tuple[Optional[redis.Redis], Optional[redis.Redis], Optional[redis.sentinel.Sentinel]]:
```
Comment (gberaudo) [inline c2cwsgiutils/redis_utils.py:32]: This looks like a breaking change.

File: c2cwsgiutils/broadcast/__init__.py [hunk 4]
```diff
@@ -6,7 +6,8 @@
 from typing import Any, Callable, Dict, List, Optional
 
 import pyramid.config
-from c2cwsgiutils import _utils, redis_utils
+
+from c2cwsgiutils import redis_utils, utils_
```
Comment (gberaudo) [inline c2cwsgiutils/broadcast/__init__.py:10]: I guess _util was used to avoid conflicting with the utils package.
What about calling it `config_util`?

**Review by gberaudo — CHANGES_REQUESTED** [review]

File: .prospector.yaml (hunk 2 above)
Comment (sbrunner) [reply .prospector.yaml:24]: No, and we can't have tow different tools to check the import order, then I disable the pylint check...

File: c2cwsgiutils/redis_utils.py (hunk 3 above)
Comment (sbrunner) [reply c2cwsgiutils/redis_utils.py:32]: +1, changing :-)

File: c2cwsgiutils/broadcast/__init__.py (hunk 4 above)
Comment (sbrunner) [reply c2cwsgiutils/broadcast/__init__.py:10]: +1 :-)

File: c2cwsgiutils/profiler.py (hunk 1 above)
Comment (sbrunner) [reply c2cwsgiutils/profiler.py:9]: `env_or_settings` works well with a `None` settings, but it's throw that it's not needed here I just do that for coherence...
What do you think?

File: c2cwsgiutils/broadcast/__init__.py (hunk 4 above)
Comment (sbrunner) [reply c2cwsgiutils/broadcast/__init__.py:10]: with a 's' to be coherent with `redis_utils`

File: c2cwsgiutils/profiler.py (hunk 1 above)
Comment (gberaudo) [reply c2cwsgiutils/profiler.py]: I would go to simplicity and only use `env_or_settings` in the cases where we actually have settings.

File: .prospector.yaml (hunk 2 above)
Comment (gberaudo) [reply .prospector.yaml:24]: Oh, great. But you also changed the import order in some files, right?

File: c2cwsgiutils/pyramid_logging.py [hunk 5]
```diff
@@ -126,7 +128,12 @@ def init(configfile: Optional[str] = None) -> Optional[str]:
         logging.config.fileConfig(configfile_, defaults=dict(os.environ))
         return configfile_
     else:
-        level = os.environ.get("LOG_LEVEL", os.environ.get("OTHER_LOG_LEVEL", "INFO"))
```
Comment (gberaudo) [inline c2cwsgiutils/pyramid_logging.py:129]: Item

File: c2cwsgiutils/redis_utils.py [hunk 6]
```diff
@@ -17,24 +18,35 @@
...
+REDIS_DB_KEY_PROP = "c2c.redis_db"
+
 _master: Optional[redis.Redis] = None
 _slave: Optional[redis.Redis] = None
 _sentinel: Optional[redis.sentinel.Sentinel] = None
 
 
-def get() -> Tuple[Optional[redis.Redis], Optional[redis.Redis], Optional[redis.sentinel.Sentinel]]:
+def get(
+    settings=None,
```
Comment (gberaudo) [inline c2cwsgiutils/redis_utils.py:33]: Is there some typing we can put here?

File: .prospector.yaml (hunk 2 above)
Comment (sbrunner) [reply .prospector.yaml:24]: Yes with isort according the config available in setup.cfg :-)

**Comment by sbrunner** [comment]
@gberaudo I do all the request changes :-)

**Review by gberaudo — APPROVED** [review]
Thanks

**Review by fredj — APPROVED** [review]
thanks @sbrunner

**Pull request merged**